training_data_loader.cpp

```diff
@@ -396,13 +397,16 @@ struct FeaturedBatchStream : Stream<StorageT>
 
 extern "C" {
 
-    EXPORT Stream<SparseBatch>* CDECL create_sparse_batch_stream(const char* feature_set_c, int concurrency, const char* filename, int batch_size, bool cyclic, bool filtered)
+    EXPORT Stream<SparseBatch>* CDECL create_sparse_batch_stream(const char* feature_set_c, int concurrency, const char* filename, int batch_size, bool cyclic, bool filtered, int random_fen_skipping)
     {
         std::function<bool(const TrainingDataEntry&)> skipPredicate = nullptr;
-        if (filtered)
+        if (filtered || random_fen_skipping)
         {
-            skipPredicate = [](const TrainingDataEntry& e){
-                return e.isCapturingMove() || e.isInCheck();
+            skipPredicate = [random_fen_skipping, filtered](const TrainingDataEntry& e){
+                std::random_device rd;
+                static thread_local std::mt19937 gen(rd()) ;
+                std::uniform_int_distribution<> distrib(0, random_fen_skipping);
```

Comment:
could use bernoulli distribution and pass a float?

Reply:
OK, done.